src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicConnection.cs

```diff
@@ -333,6 +341,11 @@ private static uint HandleEventPeerCertificateReceived(State state, ref Connecti
                 return MsQuicStatusCodes.InvalidState;
             }
 
+            if (connection._isServer)
```

Comment:
Could you add a small comment tying this `state.Connection = null;`  to the logic in inbound connection ctor?
https://github.com/dotnet/runtime/blob/a1434298983d08458e0780b83c3f803b5571fd98/src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicConnection.cs#L141-L145

It's not obvious on the first sight and without reading the PR description this would be super confusing.